lib/ansible/executor/module_common.py

```diff
@@ -1106,3 +1106,32 @@ def modify_module(module_name, module_path, module_args, templar, task_vars=None
         b_module_data = b"\n".join(b_lines)
 
     return (b_module_data, module_style, shebang)
+
+
+def get_action_args_with_defaults(action, args, defaults, templar):
+
+    tmp_args = {}
+    module_defaults = {}
+
+    # Merge latest defaults into dict, since they are a list of dicts
+    if isinstance(defaults, list):
```

Comment:
Is it even possible for this not to be a list? The original code didn't have this check.

Reply:
i made it so it could be called before/after being transformed, original code did the transformation itself in single spot ... was thinking eventually that should be the case and this should 'skip it' if what is passed was already transformed to avoid reprocessing